Modules/database_manager.py

```diff
@@ -0,0 +1,380 @@
+            connect_str = ("Driver={PostgreSQL UNICODE};"
+                           f"Server={config_['server']};"
+                           f"Port={config_['port']};"
+                           f"Database={config_['database']};"
+                           f"Uid={config_['username']};"
+                           f"Pwd={config_['password']};"
+                           "MaxVarcharSize=1024 * 1024 * 1024"
+                           # Allow bigger VarcharSize to allow faster interaction
+                           )
+        else:
+            config_ = config_[config_["method"].lower()]
+            connect_str = (f"DSN={config_['dsn']};"
+                           f"Uid={config_['username']};"
+                           f"Pwd={config_['password']};"
+                           "MaxVarcharSize=1024 * 1024 * 1024"
+                           # Allow bigger VarcharSize to allow faster interaction
+                           )
+        self.connection: pyodbc.Connection = pyodbc.connect(connect_str, autocommit=True)
+        # configure the connection
+        self.connection.setdecoding(pyodbc.SQL_WCHAR, encoding='utf-8')
+        self.connection.setencoding(encoding='utf-8')
+        self.connection.maxwrite = 1024 * 1024 * 1024  # Again, increase this for faster access
+        # and create a cursor
+        self.cursor = self.connection.cursor()
+        self._enabled = True
+        # Ensure all the default tables are defined
+        self._execute("CREATE TABLE IF NOT EXISTS fact ("
+                      "name VARCHAR NOT NULL,"
+                      "lang VARCHAR NOT NULL,"
+                      "message VARCHAR NOT NULL,"
+                      "author VARCHAR, "
+                      "PRIMARY KEY (name, lang) "
+                      #  "UNIQUE (name, lang)"
+                      ");"
+                      )
+
+        self._execute("CREATE TABLE IF NOT EXISTS fact_timestamps ("
+                      "name VARCHAR NOT NULL,"
+                      "lang VARCHAR NOT NULL,"
+                      "last_modified TIMESTAMP WITH TIME ZONE NOT NULL,"
+                      "PRIMARY KEY (name, lang)"
+                      ");"
+                      )
+
+    def __init__(self):
+
+        """
+        Connects to the DB, sets up the connection, retrieves the cursor.
+        Creates the default tables should they not exist.
+        """
+
+        # if we are not supposed to connect on start-up, we just sit tight and do nothing
+        if not self.enabled:
+            return
+        # we init the connection, as we are supposed to do
+        try:
+            self.init_connection()
+        except pyodbc.Error:
+            # something went terribly wrong and excrements hit the ventilation device
+            # using internal here to skip disconnection part as it might (and probably would) call
+            # stuff on a NoneType
+            self._enabled = False
+            return

```

Comment:
should the exception be re-raised? unless im mistaken this will silently disconnect the database
If this is intentional, we should at least log the exception